javascript/javascript3/week2/promise.js

```diff
@@ -0,0 +1,86 @@
+//Promise that resolves after set time:
+function promiseResolveAfterSetTime(resolveAfter) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, resolveAfter * 1000);
+  });
+}
+
+promiseResolveAfterSetTime(5).then(() => {
+  console.log("I am called asynchronously"); // logged out after 5 seconds
+});
+//using async await function
+
+async function usingAsyncAwait() {
+  try {
+    await promiseResolveAfterSetTime(5);
+    console.log("I am called asynchronously using async and await function");
+  } catch (error) {
+    console.log(error);
+  }
+}
+usingAsyncAwait();
+
+//Rewrite time:
+function setTimeoutPromise(resolveAfetr) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
```

Comment:
There is a small mistake. A string argument is need for resolve function.
Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?ts(2794)